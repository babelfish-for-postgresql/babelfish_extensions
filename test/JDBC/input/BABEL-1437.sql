-- Test unexist db, expecting null
SELECT db_id('hello');
SELECT db_name(1234);
GO

-- Test master, tempdb, and msdb
SELECT db_id('master');
SELECT db_name(1);
GO

USE master;
GO

SELECT db_id();
SELECT db_name();
GO

SELECT db_id('tempdb')
SELECT db_name(2);
GO

USE tempdb;
GO
SELECT db_id();
SELECT db_name();
GO

SELECT db_id('msdb')
SELECT db_name(4);
GO

USE msdb;
GO
SELECT db_id();
SELECT db_name();
GO

-- Test custom database
CREATE DATABASE babel_1437_db;
GO
USE babel_1437_db;
GO

SELECT (case when db_name() = 'babel_1437_db' then 'true' else 'false' end) result;
SELECT (case when db_name(db_id()) = 'babel_1437_db' then 'true' else 'false' end) result;
SELECT (case when db_id('babel_1437_db') = db_id() then 'true' else 'false' end) result;
GO

-- test dropped database, expecting db_id to return null
USE MASTER;
GO

DROP DATABASE babel_1437_db;
GO
SELECT db_id('babel_1437_db');
GO
