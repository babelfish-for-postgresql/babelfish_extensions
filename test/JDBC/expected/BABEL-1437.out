-- Test unexist db, expecting null
SELECT db_id('hello');
SELECT db_name(1234);
GO
~~START~~
smallint
<NULL>
~~END~~

~~START~~
nvarchar
<NULL>
~~END~~


-- Test master, tempdb, and msdb
SELECT db_id('master');
SELECT db_name(1);
GO
~~START~~
smallint
1
~~END~~

~~START~~
nvarchar
master
~~END~~


USE master;
GO

SELECT db_id();
SELECT db_name();
GO
~~START~~
smallint
1
~~END~~

~~START~~
nvarchar
master
~~END~~


SELECT db_id('tempdb')
SELECT db_name(2);
GO
~~START~~
smallint
2
~~END~~

~~START~~
nvarchar
tempdb
~~END~~


USE tempdb;
GO
SELECT db_id();
SELECT db_name();
GO
~~START~~
smallint
2
~~END~~

~~START~~
nvarchar
tempdb
~~END~~


SELECT db_id('msdb')
SELECT db_name(4);
GO
~~START~~
smallint
4
~~END~~

~~START~~
nvarchar
msdb
~~END~~


USE msdb;
GO
SELECT db_id();
SELECT db_name();
GO
~~START~~
smallint
4
~~END~~

~~START~~
nvarchar
msdb
~~END~~


-- Test custom database
CREATE DATABASE babel_1437_db;
GO
USE babel_1437_db;
GO

SELECT (case when db_name() = 'babel_1437_db' then 'true' else 'false' end) result;
SELECT (case when db_name(db_id()) = 'babel_1437_db' then 'true' else 'false' end) result;
SELECT (case when db_id('babel_1437_db') = db_id() then 'true' else 'false' end) result;
GO
~~START~~
text
true
~~END~~

~~START~~
text
true
~~END~~

~~START~~
text
true
~~END~~


-- test dropped database, expecting db_id to return null
USE MASTER;
GO

DROP DATABASE babel_1437_db;
GO
SELECT db_id('babel_1437_db');
GO
~~START~~
smallint
<NULL>
~~END~~

